api/handlers.go

```diff
@@ -158,11 +98,72 @@ func GetStatus(w http.ResponseWriter, r *http.Request) {
 	}
 
 	w.Header().Set("Content-Type", "application/json")
-	w.Write(bytes)
+	if _, err := w.Write(bytes); err != nil {
```

Comment:
I think we should check that the number of bytes written matches with the amount we want to send. If not you should attempt to resend the slice which was not sent.